### PR TITLE
Handle Project and File events on the app

### PR DESCRIPTION
**Pull request by gatzjames**
Handle more events from SSE API:
- [x] TeamProjectChangedEvent - Event for when a project gets added/renamed/deleted
- [x] FileDeletedEvent - Event for when a file gets deleted
- [x] BranchDeletedEvent - Event for when a branch gets deleted
- [x] FileChangedEvent - Event for when a file gets renamed
- [x] PresenceContext is not responsible for strictly Presence events anymore but handles more. Renamed to insomnia-event-stream

Closes INS-3225 INS-3315

### CODE REVIEW

**Review by filfreire — DISMISSED** [review]

File: packages/insomnia/src/ui/context/app/insomnia-event-stream-context.tsx [hunk 1]
```diff
@@ -111,7 +162,7 @@ export const PresenceProvider: FC<PropsWithChildren> = ({ children }) => {
 
         source.addEventListener('message', e => {
           try {
-            const presenceEvent = JSON.parse(e.data) as UserPresenceEvent;
+            const presenceEvent = JSON.parse(e.data) as UserPresenceEvent | TeamProjectChangedEvent | FileDeletedEvent | BranchDeletedEvent | FileChangedEvent;
 
             if (presenceEvent.type === 'PresentUserLeave') {
```
Comment (jackkav) [inline packages/insomnia/src/ui/context/app/insomnia-event-stream-context.tsx]: readability would be helped here by making a const for each condition eg. `const didUserLeaveOrg = presenceEvent.type === 'PresentUserLeave'`

File: packages/insomnia/src/ui/context/app/insomnia-event-stream-context.tsx [hunk 2]
```diff
@@ -101,6 +142,16 @@ export const PresenceProvider: FC<PropsWithChildren> = ({ children }) => {
         console.log('Error parsing response', e);
       }
     }
+
```
Comment (gatzjames) [inline packages/insomnia/src/ui/context/app/insomnia-event-stream-context.tsx]: We need to test this at scale

**Comment by jackkav** [comment]
Perhaps the PR title should make clear, besides handling more events this PR also:
- adds two cases to the SSE listener to response to specific events
- adds two action calls every 60 seconds which will make a number of fetches from the insomnia api to satisfy cases involving orgs and projects.

it would be useful to describe the nature of these cases since the code is not very self documenting.

**Review by filfreire — APPROVED** [review]
tested it with @gatzjames, LGTM

**Pull request merged**